demisto_sdk/commands/generate_unit_tests/generate_unit_tests.py

```diff
@@ -0,0 +1,314 @@
+        """
+        if Path(self.input_path).is_file():
+            with open(self.input_path, 'r') as input_file:
+                return input_file.read()
+        else:
+            print_error('failed to open input file')
+
+    def decision_maker(self, command_name):
+        """
+        Returns true if unit test should be generated to a command, false otherwise
+        """
+        return len(self.commands) == 0 or command_name in self.commands_to_generate
+
+    @staticmethod
+    def command_name_transformer(command_name):
+        return command_name.strip('!').replace('-', '_') + '_command'
+
+    def build_example_dict(self):
+        """
+        gets an array of command examples, run them one by one and return a map of
+            {base command -> {readable_outputs:markdown, outputs:context_outputs}}
+        Note: if a command appears more then once, run all occurrences but stores only the last.
+        """
+        examples = {}  # type: dict
+        errors = []  # type: list
+        for example in self.command_examples:
+            name, md_example, context_example, cmd_errors = execute_command(example, self.insecure)
+            if 'playbookQuery' in context_example:
+                del context_example['playbookQuery']
+
+            context_example = json.dumps(context_example)
+            errors.extend(cmd_errors)
+
+            if not cmd_errors:
+                name = self.command_name_transformer(name)
+                examples[name] = {'readable_output': md_example, 'outputs': context_example}
+        return examples, errors
+
+    def create_command_to_generate_dict(self):
+        """
+        Parses command_examples into dictionary of command name and arguments.
+        """
+        for command in self.command_examples:
+            command_line = command.split(' ')
+            command_name = self.command_name_transformer(command_line[0])
+            command_dict = {}
+            for arg in command_line[1:]:
+                key = arg.split('=')[0]
+                value = arg.split('=')[1]
+                value = value.replace('"', '')
+                command_dict.update({key: value})
+            if command_name in self.commands_to_generate:
+                self.commands_to_generate.get(command_name).append(command_dict)
+            else:
+                self.commands_to_generate.update({command_name: [command_dict]})
+        click.echo('Unit tests will be generated for the following commands:')
+        click.echo('\n'.join(self.commands_to_generate.keys()))
+
+    def run_commands(self):
+        """
+        Runs commands using Demisto instance
+        """
+        global logger
+        self.example_dict, build_errors = self.build_example_dict()
+        if build_errors:
+            logger.error('Found errors while executing command using demisto:')
+            logger.error('\n'.join(build_errors))
+
+
+class CustomContactSolver(ContractSolver):
```

Comment:
let's add docstrings to the methods of this class (including the c'tor, in which specify the attributes used in the methods)